packages/wdio-appium-service/src/launcher.ts

```diff
@@ -0,0 +1,164 @@
+            command = 'cmd'
+        }
+
+        return command
+    }
+
+    /**
+     * update capability connection options to connect
+     * to Appium server
+     */
+    private _setCapabilities() {
+        this._capabilities.forEach(
+            (cap) => !isCloudCapability(cap) && Object.assign(
+                cap,
+                DEFAULT_CONNECTION,
+                'port' in this._args ? { port: this._args.port } : {},
+                { path: this._args.basePath },
+                { ...cap }
+            ))
+    }
+
+    async onPrepare() {
+        /**
+         * Append remaining arguments
+         */
+        this._appiumCliArgs.push(...formatCliArgs(this._args))
+
+        this._setCapabilities()
+
+        /**
+         * start Appium
+         */
+        this._process = await promisify(this._startAppium)(this._command, this._appiumCliArgs)
+
+        if (this._logPath) {
+            this._redirectLogStream(this._logPath)
+        }
+    }
+
+    onComplete() {
+        if (this._process) {
+            log.debug(`Appium (pid: ${this._process.pid}) killed`)
+            this._process.kill()
+        }
+    }
+
+    private _startAppium(command: string, args: Array<string>, callback: (err: any, result: any) => void): void {
+        log.debug(`Will spawn Appium process: ${command} ${args.join(' ')}`)
+        let process: ChildProcessByStdio<null, Readable, Readable> = spawn(command, args, { stdio: ['ignore', 'pipe', 'pipe'] })
+        let error: Error | undefined
+
+        process.stdout.on('data', (data) => {
+            if (data.includes('Appium REST http interface listener started')) {
+                log.debug(`Appium started with ID: ${process.pid}`)
+                callback(null, process)
+            }
+        })
+
+        /**
+         * only capture first error to print it in case Appium failed to start.
+         */
+        process.stderr.once('data', err => { error = err })
+
+        process.once('exit', exitCode => {
+            let errorMessage = `Appium exited before timeout (exit code: ${exitCode})`
+            if (exitCode == 2) {
+                errorMessage += '\n' + (error || 'Check that you don\'t already have a running Appium service.')
+                log.error(errorMessage)
+            }
+            callback(new Error(errorMessage), null)
+        })
+    }
+
+    private _redirectLogStream(logPath: string) {
+        if (!this._process){
+            throw Error('No Appium process to redirect log stream')
+        }
+        const logFile = getFilePath(logPath, DEFAULT_LOG_FILENAME)
+
+        // ensure file & directory exists
+        ensureFileSync(logFile)
+
+        log.debug(`Appium logs written to: ${logFile}`)
+        const logStream = createWriteStream(logFile, { flags: 'w' })
+        this._process.stdout.pipe(logStream)
+        this._process.stderr.pipe(logStream)
+    }
+
+    private static _getAppiumCommand (moduleName = 'appium') {
```

Comment:
Is there a reason we moved it into the class as static method? If not shall we move it back into utils?

Reply:
yeah we discussed it here https://github.com/webdriverio/webdriverio/pull/6149#discussion_r528990943 
If you think it falls in scope utils I can move it back as well.

Reply:
Actually, I forgot to move tests to launcher.test.ts. Anyhow I need to change it

Reply:
Ah sorry about this.